src/Storages/StorageFile.cpp

```diff
@@ -1969,6 +2006,27 @@ StorageFile::ArchiveInfo StorageFile::getArchiveInfo(
     return archive_info;
 }
 
+StorageFile::ArchiveInfo StorageFile::getArchiveInfo(
+    const std::vector<std::string> & paths_to_archive,
+    const std::vector<std::string> & files_in_archive,
+    const std::string & user_files_path,
+    ContextPtr context,
+    size_t & total_bytes_to_read)
+{
+    ArchiveInfo archive_info;
+    archive_info.paths_to_archives = getPathsList(paths_to_archive, user_files_path, context, total_bytes_to_read);
+    if (!files_in_archive.empty())
+    {
+        std::unordered_set<std::string> paths_in_archive;
+        paths_in_archive.insert(files_in_archive.begin(), files_in_archive.end());
+
+        archive_info.filter
```

Comment:
Here we put all the paths into a single ArchiveInfo, which will result in us trying to find all of the files in all of the archives. For example with `['first.zip :: 1.csv', 'second.zip :: 2.csv']` we will try to find `1.csv` and `2.csv` files in both of the archives which should now happed. We also don't have support for globs with this solution, but we should.

As a solution, there should probably be separate ArchiveInfo for each of the paths that have been given to us.